scylla-cdc-replicator/src/main.rs

```diff
@@ -1,29 +1,58 @@
 mod replication_tests;
 pub mod replicator_consumer;
 
-use replicator_consumer::ReplicatorConsumerFactory;
-use scylla::SessionBuilder;
-use std::error::Error;
-use std::sync::Arc;
 use std::time::Duration;
 
+use clap::Parser;
+
+use replicator_consumer::Replicator;
+
+#[derive(Parser)]
+struct Args {
+    #[clap(short, long)]
+    keyspace: String,
+
+    #[clap(short, long)]
+    table: String,
+
+    #[clap(short, long)]
+    source: String,
+
+    #[clap(short, long)]
+    destination: String,
+
+    #[clap(long, default_value_t = 60)]
+    window_size: u16,
+
+    #[clap(long, default_value_t = 30)]
+    safety_interval: u16,
+
+    #[clap(long, default_value_t = 10)]
+    sleep_interval: u16,
+}
+
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn Error>> {
-    println!("Hello");
-    tokio::time::sleep(Duration::from_secs(1)).await;
-
-    // Create a factory to remove the warning about unused new() function.
-    let _factory = ReplicatorConsumerFactory::new(
-        Arc::new(
-            SessionBuilder::new()
-                .known_node("127.0.0.1:9042")
-                .build()
-                .await?,
-        ),
-        "ks".to_string(),
-        "t".to_string(),
-    );
-
-    println!("Replicator");
+async fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    let (mut replicator, _handle) = Replicator::new(
+        args.source,
+        args.keyspace.clone(),
+        args.table.clone(),
+        args.destination,
+        args.keyspace,
+        args.table,
+        Duration::from_secs(args.window_size as u64),
+        Duration::from_secs(args.safety_interval as u64),
+        Duration::from_secs(args.sleep_interval as u64),
+    )
+    .await?;
+
+    tokio::signal::ctrl_c().await.unwrap();
+
+    println!("Shutting down...");
+
+    replicator.stop();
```

Comment:
`stop()` does not wait for the replicator to finish, it just sets the upper timestamp so that the log reader stops after reaching it. The whole program will stop immediately if the `main` function returns, regardless of other running tasks.

If you want to gracefully wait until the replicator finishes, you should `.await` then handle returned from the replicator.